querybook/server/lib/query_review/base_query_review_handler.py

```diff
@@ -0,0 +1,146 @@
+from app.datasource import api_assert
+from const.datasources import (
+    INVALID_SEMANTIC_STATUS_CODE,
+    RESOURCE_NOT_FOUND_STATUS_CODE,
+)
+from lib.query_review.utils import (
+    notify_reviewers_of_new_request,
+)
+from logic import query_review as logic, query_execution as qe_logic
+from const.query_execution import PeerReviewParamsDict, QueryExecutionStatus
+from tasks.run_query import run_query_task
+
+
+def validate_reviewer_permission(query_review_id: int, reviewer_id: int, session):
+    """
+    Helper function to validate reviewer permissions.
+    """
+    query_review = logic.get_query_review(query_review_id, session=session)
+    api_assert(
+        query_review is not None,
+        f"QueryReview with id {query_review_id} does not exist.",
+        RESOURCE_NOT_FOUND_STATUS_CODE,
+    )
+    current_status = query_review.get_status()
+    api_assert(
+        current_status == "pending",
+        "This review has already been processed.",
+        INVALID_SEMANTIC_STATUS_CODE,
+    )
+    api_assert(
+        reviewer_id in [reviewer.id for reviewer in query_review.assigned_reviewers],
+        "You are not assigned as a reviewer for this query.",
+    )
+    return query_review
+
+
+class BaseQueryReviewHandler:
+    def initiate_query_peer_review_workflow(
+        self,
+        query_execution_id: int,
+        uid: int,
+        peer_review_params: PeerReviewParamsDict,
+        session=None,
+    ):
+        """
+        Initiates the query peer review workflow by creating a QueryReview,
+        assigning reviewers, and sending notifications.
+        """
+        reviewer_ids = peer_review_params["reviewer_ids"]
+        request_reason = peer_review_params["request_reason"]
+
+        if uid in reviewer_ids:
+            api_assert(
+                False,
+                "You cannot assign yourself as a reviewer.",
+                INVALID_SEMANTIC_STATUS_CODE,
+            )
```

Comment:
about the reviewers, something came to mind. does the data privacy side know a reviewer is eligible for a review or not?